riscv/csrs.h

```diff
@@ -61,7 +62,11 @@ typedef std::shared_ptr<csr_t> csr_t_p;
 class basic_csr_t: public csr_t {
  public:
   basic_csr_t(processor_t* const proc, const reg_t addr, const reg_t init);
-  virtual reg_t read() const noexcept override;
+
+  inline virtual reg_t read() const noexcept override {
```

Comment:
The `inline` keyword doesn't have any effect here, according to [GCC docs](https://gcc.gnu.org/onlinedocs/gcc/Inline.html): "As required by ISO C++, GCC considers member functions defined within the body of a class to be marked inline even if they are not explicitly declared with the inline keyword."

So is there any reason to specify it explicitly?

Reply:
No, you're right, I forgot about that.